### PR TITLE
make hash the same as js

### PR DESCRIPTION
**Pull request by edewit**
also part of [RHMAP-2528](https://issues.jboss.org/browse/RHMAP-2528)

make sure the hashes are the same otherwise unnecessary syncing will happen

### CODE REVIEW

**Comment by edewit** [comment]
@wei-lee sort was different on js, created very simple version with all characters on my keyboard

File: FHSDK/Sync/FHSyncUtils.cs [hunk 1]
```diff
@@ -75,7 +75,7 @@ private static JArray SortObj(object pObject)
 			} else if (jsonObj.Type == JTokenType.Object) {
 				JObject casted = jsonObj as JObject;
 				List<String> keys = casted.Properties ().Select (c => c.Name).ToList ();
-				keys.Sort ();
+				keys.Sort(Comparison);
```
Comment (wei-lee) [inline FHSDK/Sync/FHSyncUtils.cs]: how the sort works by default?

**Comment by edewit** [comment]
@wei-lee by default '_' is before all the letters

File: FHSDK/Sync/FHSyncUtils.cs (hunk 1 above)
Comment (wei-lee) [inline FHSDK/Sync/FHSyncUtils.cs]: this is fine for English. But I am just a bit concerned if the customer has a dataset using some other languages.

Comment (edewit) [inline FHSDK/Sync/FHSyncUtils.cs]: Found out it's actually based on character int value changed it accordingly

Comment (wei-lee) [inline FHSDK/Sync/FHSyncUtils.cs]: :+1:

**Comment by edewit** [comment]
@wei-lee found out that the javascript sort is based on char int value, changed it according now also works for strange languages that use something like ü, ö and ä

**Comment by wei-lee** [comment]
looks good

**Pull request merged**